dash/_validate.py

```diff
@@ -125,7 +125,7 @@ def validate_output_spec(output, output_spec, Output):
     for outi, speci in zip(output, output_spec):
         speci_list = speci if isinstance(speci, (list, tuple)) else [speci]
         for specij in speci_list:
-            if Output(specij["id"], specij["property"]) != outi:
+            if not Output(specij["id"], specij["property"]) == outi:
```

Comment:
Really hoping this is the last Py2 bug I have to fix 🤦 
We implemented [`__eq__` for dependencies](https://github.com/plotly/dash/blob/66c845126c9985bd5c96e84f9846532c659d8fd0/dash/dependencies.py#L56) but in [Py2](https://docs.python.org/2/reference/datamodel.html?highlight=__ne__#object.__ne__)
> There are no implied relationships among the comparison operators. The truth of `x==y` does not imply that `x!=y` is false.

whereas in [Py3](https://docs.python.org/3.8/reference/datamodel.html?highlight=__ne__#object.__ne__)
> For `__ne__()`, by default it delegates to `__eq__()` and inverts the result